### PR TITLE
feat: add model classes VC and VP

### LINKED ISSUE #3501: IATP: create model classes and transformers
# Feature Request

This issue is intended as preparatory work for the IATP implementations, it calls for the implementation of the following model classes:
- `VerifiableCredential`
- `VerifiablePresentation`

these go into an SPI module together with the interfaces.

Other objects we need beforehand include JSON-LD transformers, which should go in a separate module.

## Which Areas Would Be Affected?

iam

## Why Is the Feature Desired?

IATP adoption

## Solution Proposal

_If possible, provide a (brief!) solution proposal._

### PR DESCRIPTION
**Pull request by paullatzelsperger**
## What this PR changes/adds

- Adds model classes and tests for VerifiableCredentials and VerifiablePresentations. These implementations are very similar to what we previously had in the [IdentityHub](https://github.com/eclipse-edc/IdentityHub/tree/main/spi/identity-hub-spi/src/main/java/org/eclipse/edc/identityhub/spi/credentials/model), minus the JSON annotations, and a bit more restrictive.

## Why it does that

They will be the base model classes for upcoming implementations regarding IATP adoption.

## Further notes

- Adding transformer classes to and from JSON-LD will come in another PR

## Linked Issue(s)

Closes #3501 

_Please be sure to take a look at the [contributing guidelines](https://github.com/eclipse-edc/.github/blob/main/CONTRIBUTING.md#submit-a-pull-request) and our [etiquette for pull requests](https://github.com/eclipse-edc/.github/blob/main/contributing/pr_etiquette.md)._

### CODE REVIEW

File: spi/common/identity-trust-spi/src/main/java/org/eclipse/edc/identitytrust/model/Proof.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+ * </ul>
+ * The verification method <strong>cannot</strong> be anything other than URI or object.
+ * Going by the aforementioned examples, {@code jws} and {@code proofValue} would be contained in {@link Proof#proofContents}
+ */
+public class Proof {
+    private Map<String, Object> proofContents = new HashMap<>();
+    private String type;
+    private Date created;
+    private String proofPurpose;
+    private Object verificationMethod;
```
Comment (bscholtes1A) [inline spi/common/identity-trust-spi/src/main/java/org/eclipse/edc/identitytrust/model/Proof.java:70]: Map<String,Object> instead of just Object here no?

Comment (paullatzelsperger) [reply spi/common/identity-trust-spi/src/main/java/org/eclipse/edc/identitytrust/model/Proof.java:70]: the `verificationMethod` can be a `URI` (linked proof) or a `Map<String, Object>` (embedded proof)

Comment (bscholtes1A) [reply spi/common/identity-trust-spi/src/main/java/org/eclipse/edc/identitytrust/model/Proof.java:70]: Didn't know. Good for me then!

**Review by bscholtes1A — APPROVED** [review]

File: spi/common/identity-trust-spi/src/main/java/org/eclipse/edc/identitytrust/model/Proof.java (hunk 1 above)
Comment (paullatzelsperger) [reply spi/common/identity-trust-spi/src/main/java/org/eclipse/edc/identitytrust/model/Proof.java:70]: the `Proof.java` class contains examples for either in its javadoc 👍

File: spi/common/identity-trust-spi/src/main/java/org/eclipse/edc/identitytrust/model/VerifiableCredential.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    private Object issuer; // can be URI or an object containing an ID
+    private Instant issuanceDate; // v2 of the spec renames this to "validFrom"
+    private Instant expirationDate; // v2 of the spec renames this to "validUntil"
+    private CredentialStatus credentialStatus;
+
+    private VerifiableCredential(String rawVc, CredentialFormat format) {
+        this.rawVc = rawVc;
+        this.format = format;
+    }
+
```
Comment (jimmarino) [inline spi/common/identity-trust-spi/src/main/java/org/eclipse/edc/identitytrust/model/VerifiableCredential.java:41]:
```suggestion
```

**Review by jimmarino — APPROVED** [review]

**Comment by paullatzelsperger** [comment]
@bscholtes1A re-requested your review, because I changed this a little: we need to keep the raw VC/VP around for verification. The raw VC/VP can be either a JSON(-LD) String, or a JWT

**Review by wolf4ood — APPROVED** [review]

**Review by bscholtes1A — APPROVED** [review]

**Pull request merged**